src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -731,6 +731,34 @@ public ContainerCreation commitContainer(final String containerId,
   }
 
   @Override
+  public void renameContainer(final String containerId, final String name)
+          throws DockerException, InterruptedException {
+    WebTarget resource = resource()
+            .path("containers").path(containerId).path("rename");
+
+    if (name != null) {
```

Comment:
Good point. I coped that section from `createContainer` and didn't really think about it. The reason it's checked for null in `createContainer` is that trying to pass null to the regex matcher throws a `NullPointerException`, and no one wants to see that.

But what is the behavior we do want here?
* Check for `name=null` at the beginning of `renameContainer` and throw some `Exception`.
* Leave as is: check for `name=null` at the beginning of `renameContainer`, but do nothing. Let the request go through with no `name` query param and let docker return a 500 status. (This may seem crazy, but honestly, all of these methods that accept IDs and names don't check any of them for null and just pass them right on through. So this would kinda be consistent with the existing style.)
* Annotate the `name` param with `@NotNull` in the method signature. (But if we do it here, why not everywhere?)

I'm open to any of these approaches.